### PR TITLE
Clear magic numbers from code

### PR DESCRIPTION
**Pull request by mikisch81**

### CODE REVIEW

**Comment by Patater** [comment]
Thanks. This on the right track.

Please resolve test failures.

File: core/vmpu/src/vmpu.c [hunk 1]
```diff
@@ -616,10 +616,12 @@ int vmpu_init_pre(void)
 
 void vmpu_init_post(void)
 {
+#if !defined(ARCH_MPU_ARMv8M)
```
Comment (mikisch81) [inline core/vmpu/src/vmpu.c]: According to the Cortex M33 user guide, there is no NONBASETHRDENA bit in SCB->CCR (this bit is reserved):
image

File: core/debug/src/debug.c [hunk 2]
```diff
@@ -376,10 +376,12 @@ static void debug_fault_hard(void)
 
 void debug_init(void)
 {
+#if !defined(ARCH_MPU_ARMv8M)
```
Comment (mikisch81) [inline core/debug/src/debug.c]: According to the Cortex M33 user guide, there is no DISDEFWBUF bit in SCnSCB->ACTLR (this bit is reserved):
image

**Comment by mikisch81** [comment]
@Patater @niklas-arm @alzix 
Can you take another look, seems that there were 2 register sets which are not right for v8m.

File: core/vmpu/src/vmpu.c (hunk 1 above)
Comment (Patater) [reply core/vmpu/src/vmpu.c]: Woah. This is a big concern. If it turns out we can't do `NONBASETHRDENA` on M33, uVisor has to be reengineered a bit. We might come off lucky if `NONBASETHRDENA` is always effectively on by default, though.

File: core/debug/src/debug.c (hunk 2 above)
Comment (Patater) [reply core/debug/src/debug.c]: Without this, I wonder if we still get good debug information from the MMFAR or SFAR.

File: core/vmpu/src/vmpu.c (hunk 1 above)
Comment (Patater) [reply core/vmpu/src/vmpu.c]: Looks like the NONBASETHRDENA bit is always 1, so we are good to go.

Comment (mikisch81) [reply core/vmpu/src/vmpu.c]: But bit 0 is not NONBASETHRDENA in cortex m33, so how can you have this functionality in v8m?

**Comment by Patater** [comment]
Merged manually. Thanks!

**Pull request closed without merging**